virtualBridge.py

```diff
@@ -70,6 +74,56 @@ def setup_dhcp_entry(self, cfg: NodeConfig) -> None:
         cmd = f"virsh net-update default add ip-dhcp-host \"{host_xml}\" --live --config"
         self.hostconn.run_or_die(cmd)
 
+    def remove_dhcp_entries(self, vms: list[NodeConfig]) -> None:
+        def filter_dhcp_leases(j: list[dict[str, str]], removed_macs: list[str], names: list[str]) -> list[dict[str, str]]:
+            filtered = []
+            for entry in j:
+                if entry["mac-address"] in removed_macs:
+                    logger.info(f'Removed host with mac {entry["mac-address"]}')
+                    continue
+                if "hostname" in entry and entry["hostname"] in names:
+                    logger.info(f'Removed host with name {entry["hostname"]}')
+                    continue
+                logger.info(f'Kept entry {entry}')
+                filtered.append(entry)
+            return filtered
+
+        xml_str = self.hostconn.run("virsh net-dumpxml default").out
+        q = et.fromstring(xml_str)
+        removed_macs = []  # type: list[str]
```

Comment:
is this annotation necessary? In any case, it seems nicer to have type annotations in code instead of comments. `removed_macs: list[str] = []`

Reply:
Agreed. Will update PR.